autogluon/utils/tabular/features/generators/pipeline.py

```diff
@@ -108,9 +108,9 @@ def print_feature_metadata_info(self, log_level=20):
             self._log(log_level, f'\t\tThese features do not need to be present at inference time for this FeatureGenerator.')
         if self._feature_metadata_in_unused.get_features():
             self._log(log_level, f'\tUnused Original Features (Count: {len(self._feature_metadata_in_unused.get_features())}): {self._feature_metadata_in_unused.get_features()}')
-            self._log(log_level, f'\t\tThese features were not valid input to any of the feature generators. Add a feature generator compatible with these features to utilize them.')
+            self._log(log_level, f'\t\tThese features were not used to generate any of the output features. Add a feature generator compatible with these features to utilize them.')
             self._log(log_level, f'\t\tThese features do not need to be present at inference time for this FeatureGenerator.')
-            self._feature_metadata_in_unused.print_feature_metadata_full(self.log_prefix + '\t', log_level=log_level)
+            self._feature_metadata_in_unused.print_feature_metadata_full(self.log_prefix + '\t\t', log_level=log_level)
         self._log(20, '\tOriginal Features (exact raw dtype, raw dtype):')
```

Comment:
Upon playing with the new feature generators more, I think users will find these outputs confusing:

```
Original Features (exact raw dtype, raw dtype):
            ('int64', 'int')     : 6 | ['age', 'fnlwgt', 'education-num', ...]
            ('object', 'object') : 8 | ['workclass', 'education', 'marital-status', ...]
    Original Features (raw dtype, special dtypes):
            ('int', [])    : 6 | ['age', 'fnlwgt', 'education-num', ...]
            ('object', []) : 8 | ['workclass', 'education', 'marital-status', ...]
    Processed Features (exact raw dtype, raw dtype):
            ('category', 'category') : 8 | ['workclass', 'education', 'marital-status', ...]
            ('int64', 'int')         : 6 | ['age', 'fnlwgt', 'education-num', ...]
    Processed Features (raw dtype, special dtypes):
            ('category', []) : 8 | ['workclass', 'education', 'marital-status', ...]
            ('int', [])      : 6 | ['age', 'fnlwgt', 'education-num', ...]
```

and think this could be clarified via a print-structure like this:
```
Types of features in original data:
...

Types of features in preprocessed data:
...
```

Also the reasoning behind currently presenting the types in terms of pairs `(a, b)` seems unclear from user's perspective (could instead say eg: "Originally of type a, will be treated as type b"). 

It also may be confusing that  'special dtypes' can corresponds to empty list, isn't it better just to not print anything if there's no special dtype?

Also only listing up to 3 features of each type seems quite few (and almost not worth the print statement at all for such little information).

Reply:
I also think this statement may confuse users: 
```
AutoMLPipelineFeatureGenerator processed 499 data points with 14 features
```
when the numbers don't match the earlier output `Train Data Rows:    500`

I'd vote to suppress this statement in general. If we want to print something here, I'd only do it whenever these values don't match the original # of train data rows/columns, and would also provide possible explanations (eg. datapoints with rare class-label may be deleted).

Reply:
Reworked this logging somewhat, and removed the statement you mentioned as confusing since it is redundant.